lib/archethic/account/mem_tables/uco_ledger.ex

```diff
@@ -190,4 +244,57 @@ defmodule Archethic.Account.MemTables.UCOLedger do
         end)
     end
   end
+
+  @doc """
+  Spend all the unspent outputs for the given address.
+  """
+  @spec spend_all_unspent_outputs(address :: binary(), from_init? :: boolean()) :: :ok
+  def spend_all_unspent_outputs(address, from_init? \\ false) do
+    case :ets.lookup(@unspent_output_index_table, address) do
+      [] ->
+        :ok
+
+      [{^address, true}] ->
+        :ets.delete(@unspent_output_index_table, address)
+        :ok
+
+      utxos ->
+        write_to_db(address, utxos, from_init?)
+    end
+  end
+
+  defp write_to_db(address, inputs, false) do
+    {:ok, pid} = DB.start_inputs_writer(:UCO, address)
+
+    Enum.each(inputs, fn {to, from} ->
+      [{_, amount, _, timestamp, reward?, protocol_version}] =
+        :ets.lookup(@ledger_table, {to, from})
+
+      DB.append_input(pid, %VersionedTransactionInput{
+        protocol_version: protocol_version,
+        input: %TransactionInput{
+          from: from,
+          amount: amount,
+          spent?: true,
+          reward?: reward?,
+          timestamp: timestamp,
+          type: :UCO
+        }
+      })
+
+      :ets.delete(@ledger_table, to)
+    end)
+
+    :ets.delete(@unspent_output_index_table, address)
+
+    DB.stop_inputs_writer(pid)
+  end
+
+  defp write_to_db(address, inputs, true) do
+    Enum.each(inputs, fn {to, _from} ->
+      :ets.delete(@ledger_table, to)
```

Comment:
The key must be `{to, _from}` causing the non removal of the inputs as mentioned in the screenshots

Reply:
:ok